src/Agent/NewRelic/Profiler/Configuration/InstrumentationConfiguration.h

```diff
@@ -45,52 +45,82 @@ namespace NewRelic { namespace Profiler { namespace Configuration
         }
 
         InstrumentationConfiguration(InstrumentationPointSetPtr instrumentationPoints) :
-            _instrumentationPoints(instrumentationPoints),
+            _instrumentationPointsMap(new InstrumentationPointMap()),
```

Comment:
Instead of dynamically allocating the InstrumentationPointMap here, we should probably use a field initializer without dynamic allocation to define the map.

Reply:
Was able to do this for the map, but reintroduced the set and that could not be done b/c a different constructor can send it in.